robotbase/src/main/java/frc/robot/commands/drive/DriveToApriltagCommand.java

```diff
@@ -0,0 +1,96 @@
+package frc.robot.commands.drive;
+
+import edu.wpi.first.math.controller.PIDController;
+import edu.wpi.first.math.filter.Debouncer;
+import edu.wpi.first.math.filter.Debouncer.DebounceType;
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.wpilibj2.command.Command;
+import frc.robot.Constants;
+import frc.robot.Robot;
+import frc.robot.util.Utility;
+
+public class DriveToApriltagCommand extends Command {
+  private double m_tyOffset;
+  private double m_rotationGoal;
+  private int m_pipelineIndex;
+  private Debouncer m_canSeePieceDebouncer;
+
+  private PIDController m_xController;
+  private PIDController m_yController;
+  private PIDController m_rotationController;
+
+  public DriveToApriltagCommand(double tyOffset, double rotationGoal, int pipelineIndex) {
+    m_tyOffset = tyOffset;
+    m_rotationGoal = rotationGoal;
+    m_pipelineIndex = pipelineIndex;
+
+    m_xController = Constants.SWERVE.APRILTAG_X_TRANSLATION_PID_CONTROLLER;
+    m_yController = Constants.SWERVE.APRILTAG_Y_TRANSLATION_PID_CONTROLLER;
+    m_rotationController = Constants.SWERVE.APRILTAG_ROTATION_PID_CONTROLLER;
+
+    addRequirements(Robot.swerve);
+  }
+
+  @Override
+  public void initialize() {
+    Robot.shooterLimelight.setPipeline(m_pipelineIndex);
+
+    // reset pids
+    m_yController.setSetpoint(m_tyOffset + Constants.SWERVE.APRILTAG_TY_MAGIC_OFFSET);
+    m_yController.reset();
+
+    m_xController.setSetpoint(0);
```

Comment:
Good for now, we will probably want to set something up to determine if we are tracking to the source or amp and set setpoints accordingly.